### PR TITLE
Adding the redhat service

### PR DESCRIPTION
**Pull request by fao89**
It manages RH subscription and Insights
https://linux-system-roles.github.io/rhc/

Depends-On: https://github.com/openstack-k8s-operators/edpm-ansible/pull/837
Depends-On: https://github.com/openstack-k8s-operators/openstack-operator/pull/1221

closes OSPRH-11938

### CODE REVIEW

**Review by jistr — APPROVED** [review]

File: docs_user/modules/proc_adopting-compute-services-to-the-data-plane.adoc [hunk 1]
```diff
@@ -259,6 +252,9 @@ spec:
       - ctlplane
   preProvisioned: true
   services:
+ifeval::["{build}" == "downstream"]
+    - redhat
```
Comment (bogdando) [inline docs_user/modules/proc_adopting-compute-services-to-the-data-plane.adoc:256]: what about tests, shall we add it?
please submit follow up

Comment (fao89) [reply docs_user/modules/proc_adopting-compute-services-to-the-data-plane.adoc:256]: we should update downstream CI to use this service

File: docs_user/modules/proc_adopting-compute-services-to-the-data-plane.adoc [hunk 2]
```diff
@@ -437,6 +439,9 @@ external_ids        : {hostname=standalone.localdomain, ovn-bridge=br-int, ovn-b
 $ oc patch osdpns/openstack --type=merge --patch "
 spec:
   services:
+ifeval::["{build}" == "downstream"]
+    - redhat
```
Comment (fao89) [inline docs_user/modules/proc_adopting-compute-services-to-the-data-plane.adoc:443]: adding the service to handle the subscription-manager

File: docs_user/modules/proc_adopting-networker-services-to-the-data-plane.adoc [hunk 3]
```diff
@@ -83,13 +86,23 @@ spec:
...
+ifeval::["{build}" == "downstream"]
+        rhc_release: 9.2
+        rhc_repositories:
+            - {name: "*", state: disabled}
+            - {name: "rhel-9-for-x86_64-baseos-eus-rpms", state: enabled}
+            - {name: "rhel-9-for-x86_64-appstream-eus-rpms", state: enabled}
+            - {name: "rhel-9-for-x86_64-highavailability-eus-rpms", state: enabled}
+            - {name: "openstack-17.1-for-rhel-9-x86_64-rpms", state: enabled}
+            - {name: "fast-datapath-for-rhel-9-x86_64-rpms", state: enabled}
+            - {name: "openstack-dev-preview-for-rhel-9-x86_64-rpms", state: enabled}
```
Comment (fao89) [inline docs_user/modules/proc_adopting-networker-services-to-the-data-plane.adoc:104]: enabling subscription-manager repos

**Pull request merged**